galsim/config/noise.py

```diff
@@ -345,10 +345,13 @@ def addNoise(self, config, base, im, rng, current_var, draw_method, logger):
                 # back off.
                 im -= total_sky
         else:
-            im += extra_sky
             # Do the normal PoissonNoise calculation.
-            im.addNoise(galsim.PoissonNoise(rng))
-            im -= extra_sky
+            if isinstance(total_sky, galsim.Image):
```

Comment:
Just a (minor) efficiency improvement.  If the sky is just a number (which is a typical use case), then the whole calculation can be done in one pass over the pixels, rather than adding the sky, generating noise, and subtracting, which is 3 passes.